src/common/hooks/useGetSurplusFiatValue.ts

```diff
@@ -33,10 +36,29 @@ export function useGetSurplusData(order: Order | ParsedOrder | undefined): Outpu
   const surplusFiatValue = useCoingeckoUsdValue(surplusAmount)
   const showFiatValue = Number(surplusFiatValue?.toExact()) >= MIN_FIAT_SURPLUS_VALUE
 
+  const showSurplus = shouldShowSurplus(surplusFiatValue, surplusAmount)
+
   return {
     surplusFiatValue,
     showFiatValue,
     surplusToken,
     surplusAmount,
+    showSurplus,
+  }
+}
+
+function shouldShowSurplus(
+  fiatAmount: Nullish<CurrencyAmount<Currency>>,
+  surplusAmount: Nullish<CurrencyAmount<Currency>>
+): boolean | null {
+  if (fiatAmount) {
+    // When there's a fiat amount, use that to decide whether to display the modal
+    return Number(fiatAmount.toFixed(3)) > MIN_FIAT_SURPLUS_VALUE_MODAL
```

Comment:
why converting to number? is not enough to compare directly the underling number without doing the toFixed?

Reply:
`toFixed` returns a string.
`fiatAmount` is a `Fraction` so it cannot be directly compared with a `number` afaict.

See https://docs.uniswap.org/sdk/core/reference/classes/Fraction#greaterthan

Reply:
you can compare a fraction and a fraction, 0.01 is 1/100 :P

Reply:
anyways, merging to close the release

Reply:
Well, that's true...
If you mind I can create a new PR addressing this.
Let me know